chingu-x/chingu-dashboard-be
# Discord OAuth part 1: Schema change, extended prisma client, discord oauth stubs

# Description

### Part 1 of discord oauth: 
1. Schema change (fixed seed and tests which reference the old schema)
2. extended prisma client to make it reusable for future oauth providers, we can also add more "extensions" in future now this is setup 
3. Setup passport discord strategy and guard, and stubs, console.logs will be removed in the next part

### To test this PR
1. obtain discord oauth client id and client secret (on clickup, you can also make your own for dev), and put them in .env
2. `npx prisma db push` for both dev and test db
3. go to `http://localhost:8000/api/v1/auth/discord/login` a discord authorization table will pop up, and it should return the user's basic discord profile 

## Issue link

[Clickup Link](https://app.clickup.com/t/86azr0q2z)

## Type of change



# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration


# Checklist:
